### PR TITLE
Add filename to FileNotFoundExceptions thrown in various filesystem implementations

### PR DESCRIPTION
**Pull request by Pwootage**
For #1563

### CODE REVIEW

File: src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ trait FileInputStreamFileSystem extends InputStreamFileSystem {
     case file if file.exists() && file.isFile => Array(file.getName)
     case directory if directory.exists() && directory.isDirectory && directory.list() != null =>
       directory.listFiles().map(file => if (file.isDirectory) file.getName + "/" else file.getName)
-    case _ => throw new io.FileNotFoundException("no such file or directory")
+    case _ => throw new io.FileNotFoundException(path + " (no such file or directory)")
```
Comment (dgelessus) [inline src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala]: How about changing the format to `"no such file or directory: " + path`? That way you can see the failure reason right away even if the path is very long, which I find more useful than putting the error message after the path (where it's easy to miss too). That's how Python does it for example:

``` python
>>> open("/usr/bin/nope")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/nope'
```

Same goes for `"read-only filesystem"` down below.

(Or are these exceptions that stay inside Scala and this is an established convention there? In that case feel free to ignore this.)

Comment (fnuecke) [inline src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala]: I second that, makes sense and for more readable errors. Other than that looking good!

**Comment by Pwootage** [comment]
Updated pull request with new messages as requested

**Comment by fnuecke** [comment]
Thanks!

File: src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala [hunk 2]
```diff
@@ -38,7 +38,7 @@ trait FileInputStreamFileSystem extends InputStreamFileSystem {
     case file if file.exists() && file.isFile => Array(file.getName)
     case directory if directory.exists() && directory.isDirectory && directory.list() != null =>
       directory.listFiles().map(file => if (file.isDirectory) file.getName + "/" else file.getName)
-    case _ => throw new io.FileNotFoundException("no such file or directory")
+    case _ => throw new io.FileNotFoundException("No such file or directory: " + path)
```
Comment (makkarpov) [inline src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala:41]: In Lua (as this exception will propagate to it) error messages usually start with lower case letter.

Comment (fnuecke) [inline src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala:41]: Yeah, I'll change it back to lowercase later :P

Comment (Pwootage) [inline src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala:41]: Poor non-lua developers seeing non-standard lower-case starting error
messages ;)

On Sun, Dec 20, 2015 at 3:37 PM, fnuecke notifications@github.com wrote:

> In src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala
> https://github.com/MightyPirates/OpenComputers/pull/1570#discussion_r48109340
> :
> 
> > @@ -38,7 +38,7 @@ trait FileInputStreamFileSystem extends InputStreamFileSystem {
> >      case file if file.exists() && file.isFile => Array(file.getName)
> >      case directory if directory.exists() && directory.isDirectory && directory.list() != null =>
> >        directory.listFiles().map(file => if (file.isDirectory) file.getName + "/" else file.getName)
> > -    case _ => throw new io.FileNotFoundException("no such file or directory")
> > -    case _ => throw new io.FileNotFoundException("No such file or directory: " + path)
> 
> Yeah, I'll change it back to lowercase later :P
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MightyPirates/OpenComputers/pull/1570/files#r48109340
> .

## 

Sé onr sverdar sitja hvass!
-Pwootage

**Pull request merged**